build-tools/packages/build-tools/src/fluidBuild/tasks/task.ts

```diff
@@ -22,7 +22,7 @@ export interface TaskExec {
 
 export abstract class Task {
 	public dependentTasks?: Task[];
-	private _transitiveDependentLeafTasks?: LeafTask[];
+	private _transitiveDependentLeafTasks: LeafTask[] | undefined | null;
```

Comment:
What is null needed?

Reply:
null is used to detect circular reference.